src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/MstatObjectDumper.cs

```diff
@@ -71,7 +76,29 @@ protected override void DumpObjectNode(NodeFactory factory, ObjectNode node, Obj
                 case MethodExceptionHandlingInfoNode ehInfoNode:
                     _methodEhInfo.Add(ehInfoNode.Method, objectData.Data.Length);
                     break;
+                case FieldRvaDataNode rvaDataNode:
+                    _fieldRvas.OpCode(ILOpCode.Ldtoken);
+                    _fieldRvas.Token(_emitter.EmitMetadataHandleForTypeSystemEntity(rvaDataNode.Field));
+                    _fieldRvas.LoadConstantI4(rvaDataNode.Field.GetFieldRvaData().Length);
+                    _fieldRvas.LoadConstantI4(AppendMangledName(DependencyNodeCore<NodeFactory>.GetNodeName(node, factory)));
+                    // Breakdown of RVA data was introduced in MSTAT 2.1. Readers of 2.0 should still see it in the
+                    // global blobs section. We can remove it from there in 3.0.
+                    if (VersionMajor == 2)
+                        goto reportAsBlob;
+                case ArrayOfFrozenObjectsNode:
+                    DumpFrozenRegion(factory);
+                    // Breakdown of frozen regions was introduced in MSTAT 2.1. Readers of 2.0 should still see it in the
+                    // global blobs section. We can remove it from there in 3.0.
+                    if (VersionMajor == 2)
+                        goto reportAsBlob;
+                case ResourceDataNode resourceData:
+                    DumpResourceData(factory, resourceData);
+                    // Breakdown of resource data was introduced in MSTAT 2.1. Readers of 2.0 should still see it in the
+                    // global blobs section. We can remove it from there in 3.0.
```

Comment:
`FrozenObjectNode` is not an `ObjectNode` (it doesn't carry data). `MstatObjectDumper` will only see `ObjectNode`s, so no need to handle it here.

We emit the `FrozenObjectNode` logical contents as part of the singleton `ArrayOfFrozenObjectsNode`. We need all of these to be grouped together and this is the easiest to achieve that. I admit it's a bit weird and it would be nicer if `FrozenObjectNode` also generated it's content in the object writer.